node-plugin/lang/src/context.rs

```diff
@@ -27,3 +27,14 @@ impl<'a, T: q::Text<'a>> Context<'a, T> {
         })
     }
 }
+
+impl<'a, T: q::Text<'a> + Clone> Clone for Context<'a, T> {
+    fn clone(&self) -> Self {
+        Self {
+            captures: Captures::new(),
```

Comment:
This `new` is concerning, and checking into it I think I see a bug in the Agora "interpreter".

We should clearly indicate which memory is scratch, and which is a necessary part of the query. Scratch memory should be cleared at specific points. I think right now you might be able to do funky things like use a value captured in a previous statement. I'm going to try to surface this in a test.

Even so, this `new` is still concerning within a clone. It can lead to surprises. Whether or not the contents of the captures are important depends on when the clone occurs. (Eg: If the clone occurred for some reason after match but before expression evaluation this would lead to a bug where captures were lost).

Not saying that we would run into these, but it's better to clearly indicate that this is not a direct clone. Maybe by naming it `clean_clone` or something. What do you think? Am I overreacting here?

Reply:
Fortunately, there was code to clear the captures in the right place. I had missed it in the initial search.

The point around the naming of `clone` still stands. IMO this is a common problem around clone anyway, since ambiguity abounds in whether a clone structurally shares data or has interior mutability and what the semantics of a clone actually are for a given type when used generically. So, I think this is fine.